src/sounds.cpp

```diff
@@ -388,16 +393,790 @@ std::pair<std::vector<tripoint>, std::vector<tripoint>> sounds::get_monster_soun
     return { sound_locations, cluster_centroids };
 }
 
-std::string sounds::sound_at( const tripoint &location )
-{
+std::string sounds::sound_at( const tripoint &location ) {
     auto this_sound = sound_markers.find( location );
     if( this_sound == sound_markers.end() ) {
         return std::string();
     }
-
     if( !this_sound->second.description.empty() ) {
         return this_sound->second.description;
     }
+    return _( "a sound" );
+}
+
+//sfx::
+void sfx::load_sound_effects( JsonObject &jsobj ) {
+    set_group_channels( 2, 9, 1 );
+    set_group_channels( 0, 1, 2 );
+    set_group_channels( 11, 14, 3 );
+    set_group_channels( 15, 17, 4 );
+    int index = 0;
+    std::string file;
+    sound_effect new_sound_effect;
+    new_sound_effect.id = jsobj.get_string( "id" );
+    new_sound_effect.volume = jsobj.get_int( "volume" );
+    new_sound_effect.variant = jsobj.get_string( "variant" );
+    JsonArray jsarr = jsobj.get_array( "files" );
+    while( jsarr.has_more() ) {
+        new_sound_effect.files.push_back( jsarr.next_string().c_str() );
+        file = new_sound_effect.files[index];
+        index++;
+        std::string path = ( FILENAMES[ "datadir" ] + "/sound/" + file );
+        Mix_Chunk *loaded_chunk = Mix_LoadWAV( path.c_str() );
+        if( !loaded_chunk ) {
+            dbg( D_ERROR ) << "Failed to load audio file " << path << ": " << Mix_GetError();
+        }
+        new_sound_effect.chunk = loaded_chunk;
+        sound_effects_p.push_back( new_sound_effect );
+    }
+}
+
+int sfx::get_channel( Mix_Chunk * effect_to_play ) {
+    return Mix_PlayChannel( -1, effect_to_play, 0 );
+}
+
+Mix_Chunk *do_pitch_shift( Mix_Chunk *s, float pitch ) {
+    Mix_Chunk *result;
+    Uint32 s_in = s->alen / 4;
+    Uint32 s_out = ( Uint32 )( ( float )s_in * pitch );
+    float pitch_real = ( float )s_out / ( float )s_in;
+    Uint32 i, j;
+    result = ( Mix_Chunk * )malloc( sizeof( Mix_Chunk ) );
+    result->allocated = 1;
+    result->alen = s_out * 4;
+    result->abuf = ( Uint8* )malloc( result->alen * sizeof( Uint8 ) );
```

Comment:
I see `malloc` (twice actually), but I don't see any call to `free`. \*insert quote about memory leak\*